test/LocalVocabTest.cpp

```diff
@@ -378,3 +384,27 @@ TEST(LocalVocab, getBlankNodeIndex) {
   BlankNodeIndex b = v.getBlankNodeIndex(&bnm);
   EXPECT_NE(a, b);
 }
+
+// _____________________________________________________________________________
+TEST(LocalVocab, memoryLimit) {
+  ad_utility::detail::AllocationMemoryLeftThreadsafe smallLimit =
+      ad_utility::makeAllocationMemoryLeftThreadsafeObject(
+          ad_utility::MemorySize::kilobytes(1));
+  LocalVocab localVocab(smallLimit);
+  TestWords testWords = getTestCollectionOfWords(1000);
+
+  try {
+    for (const auto& word : testWords) {
+      localVocab.getIndexAndAddIfNotContained(word);
+    }
+  } catch (const std::exception& e) {
+    std::string errorMessage = e.what();
+    EXPECT_THAT(errorMessage, ::testing::StartsWith("Tried to allocate"));
+  }
```

Comment:
We have a `AD_EXPECT_THROW_WITH_MESSAGE` macro that can be used to make this test simpler.